conda-store/Dockerfile

```diff
@@ -19,12 +21,14 @@ ENV PATH=/opt/conda/condabin:/opt/conda/envs/conda-store/bin:/opt/conda/bin:/usr
 
 COPY ./ /opt/conda-store/
 
+FROM base as dev
 RUN cd /opt/conda-store && \
     pip install -e .
+USER conda-store
+WORKDIR /opt/jupyterhub
 
-RUN mkdir -p /opt/jupyterhub && \
-    chown -R 1000:1000 /opt/jupyterhub
-
-USER 1000:1000
-
+FROM base as prod
+RUN cd /opt/conda-store && \
+    pip install .
```

Comment:
I'd prefer installing directly from PyPA than from the repository.

Reply:
@trallard Do you mean to install the conda-store package from pypi when packing the docker release?

right now, docker images are built and pushed on every merge to main, and on every release. Using PyPi would mean that we should split the workflow, so pushes to main used this repo (since it isn't pushed to PyPi), and releases installed from PyPi, which would add a dependency in the order that the jobs run.

I am fine with that if we want to do it that way, I just want to make sure we are on the same page here.